doc/reference/usermode/syscalls.rst

```diff
@@ -174,11 +174,13 @@ prototype for the implementation function is also automatically generated.
 
 The final layer is the invocation of the system call itself. All architectures
 implementing system calls must implement the seven inline functions
-:c:func:`_arch_syscall_invoke0` through :c:func:`_arch_syscall_invoke6`.  These
+:c:func:`_arch_syscall_invoke0` through :c:func:`_arch_syscall_invoke6`. These
 functions marshal arguments into designated CPU registers and perform the
-necessary privilege elevation. In this layer, all arguments are treated as an
-unsigned 32-bit type. There is always a 32-bit unsigned return value, which
-may or may not be used.
+necessary privilege elevation. Parameters of API inline function, before being
+passed as arguments to system call, are C casted to unsigned 32-bit integers,
```

Comment:
this is a bit out of date. we hard-cast everything to `uintptr_t` which matches the size of registers. only on 32-bit systems are 64-bit parameters split.

Reply:
OK, will fix that.

Reply:
Done.